demo/tree/client-tree.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'client-side-tree-demo',
+  template: `
+    <div>
+      <h3>
+        Flex Column Width Distribution
+        <small>
+          <a href="https://github.com/swimlane/ngx-datatable/blob/master/demo/columns/column-flex.component.ts" target="_blank">
+            Source
+          </a>
+        </small>
+      </h3>
+      <ngx-datatable
+        class="material"
+        [columnMode]="'flex'"
+        [headerHeight]="50"
+        [footerHeight]="50"
+        [rowHeight]="'auto'"
+        [treeFromRelation]="'manager'"
+        [treeToRelation]="'name'"
+        [rows]="rows"
+        (treeAction)="onTreeAction($event)">
+        <ngx-datatable-column name="Name" [flexGrow]="3" [isTreeColumn]="true">
+          <ng-template let-value="value" ngx-datatable-cell-template>
+            {{value}}
+          </ng-template>
+
+          <ng-template ngx-datatable-cell-tree-expander>
```

Comment:
Im torn over the individual templates for each one. I feel like it would be more straight forward to just have 1 with a state passed. IDK, maybe I'm trying to make it too abstract tho.